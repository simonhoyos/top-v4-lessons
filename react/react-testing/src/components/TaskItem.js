import React from 'react';

class TaskItem extends React.Component {
  render() {
    const { title, done, onClick } = this.props;
    return (
      <div className="task">
        <h2>{title}</h2>
        <p>{done ? 'completada' : 'por completar'}</p>
        <button
          data-testid="complete-task"
          onClick={onClick}
          type="button"
        >
          Completar
        </button>
      </div>
    );
  }
}

export default TaskItem;
