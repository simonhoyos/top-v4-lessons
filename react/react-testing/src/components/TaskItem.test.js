import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import TaskItem from './TaskItem';

describe('TaskItem' , () => {
  it('should match snapshot', () => {
    const title = 'nueva tarea';
    const done = false;
    const wrapper = shallow(<TaskItem
      done={done}
      title={title}
    />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should show completed task when task is done', () => {
    const title = 'nueva tarea';
    const done = true;
    const wrapper = shallow(<TaskItem
      done={done}
      title={title}
    />);

    expect(wrapper.find('p').text()).toMatch('completada');
  })
});
