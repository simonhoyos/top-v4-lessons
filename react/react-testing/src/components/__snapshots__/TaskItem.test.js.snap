// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskItem should match snapshot 1`] = `
<div
  className="task"
>
  <h2>
    nueva tarea
  </h2>
  <p>
    por completar
  </p>
  <button
    data-testid="complete-task"
    type="button"
  >
    Completar
  </button>
</div>
`;
